### PR TITLE
Add documentation for new .NET 8 diagnostic configuration

### PR DESCRIPTION
**Pull request by tommcdon**
## Summary

Add documentation for new .NET 8 diagnostic configuration

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/docker/build-container.md](https://github.com/dotnet/docs/blob/b48695013a54210ba6a5ada607a077ef7d18a898/docs/core/docker/build-container.md) | [Tutorial: Containerize a .NET app](https://review.learn.microsoft.com/en-us/dotnet/core/docker/build-container?branch=pr-en-us-38012) |
| [docs/core/tools/dotnet-environment-variables.md](https://github.com/dotnet/docs/blob/b48695013a54210ba6a5ada607a077ef7d18a898/docs/core/tools/dotnet-environment-variables.md) | [.NET environment variables](https://review.learn.microsoft.com/en-us/dotnet/core/tools/dotnet-environment-variables?branch=pr-en-us-38012) |

### CODE REVIEW

**Comment by tommcdon** [comment]
FYI @richlander

File: docs/core/docker/build-container.md [hunk 1]
```diff
@@ -332,7 +332,7 @@ The `WORKDIR` command changes the **current directory** inside of the container
 The next command, `ENTRYPOINT`, tells Docker to configure the container to run as an executable. When the container starts, the `ENTRYPOINT` command runs. When this command ends, the container will automatically stop.
 
 > [!TIP]
-> For added security, you can opt out of the diagnostic pipeline. When you opt-out this allows the container to run as read-only. To do this, specify a `DOTNET_EnableDiagnostics` environment variable as `0` (just before the `ENTRYPOINT` step):
+> Previous to .NET 8, containers configured to run as read-only may fail with `Failed to create CoreCLR, HRESULT: 0x8007000E`. To address this issue, specify a `DOTNET_EnableDiagnostics` environment variable as `0` (just before the `ENTRYPOINT` step):
```
Comment (IEvangelist) [inline docs/core/docker/build-container.md:335]:
```suggestion
> Before .NET 8, containers configured to run as read-only may fail with `Failed to create CoreCLR, HRESULT: 0x8007000E`. To address this issue, specify a `DOTNET_EnableDiagnostics` environment variable as `0` (just before the `ENTRYPOINT` step):
```

File: docs/core/tools/dotnet-environment-variables.md [hunk 2]
```diff
@@ -169,6 +169,24 @@ Configures the runtime to pause during startup and wait for the _Diagnostics IPC
 
 When set to `1`, enables debugging, profiling, and other diagnostics via the [Diagnostic Port](../diagnostics/diagnostic-port.md). Defaults to 1.
 
+### `DOTNET_EnableDiagnostics_IPC`
+
+> [!NOTE]
+> Available since .NET 8
+When set to `1`, enables the [Diagnostic Port](../diagnostics/diagnostic-port.md). Defaults to 1.
```
Comment (IEvangelist) [inline docs/core/tools/dotnet-environment-variables.md:176]:
```suggestion
Starting with .NET 8, when set to `1`, enables the [Diagnostic Port](../diagnostics/diagnostic-port.md). Defaults to 1.
```

File: docs/core/tools/dotnet-environment-variables.md [hunk 3]
```diff
@@ -169,6 +169,24 @@ Configures the runtime to pause during startup and wait for the _Diagnostics IPC
...
+
+### `DOTNET_EnableDiagnostics_Debugger`
+
+> [!NOTE]
+> Available since .NET 8
+When set to `1`, enables debugging. Defaults to 1.
```
Comment (IEvangelist) [inline docs/core/tools/dotnet-environment-variables.md:182]:
```suggestion
Starting with .NET 8, when set to `1`, enables debugging. Defaults to 1.
```

File: docs/core/tools/dotnet-environment-variables.md [hunk 4]
```diff
@@ -169,6 +169,24 @@ Configures the runtime to pause during startup and wait for the _Diagnostics IPC
...
+
+### `DOTNET_EnableDiagnostics_Profiler`
+
+> [!NOTE]
+> Available since .NET 8
+When set to `1`, enables profiling. Defaults to 1.
```
Comment (IEvangelist) [inline docs/core/tools/dotnet-environment-variables.md:188]:
```suggestion
Starting with .NET 8, when set to `1`, enables profiling. Defaults to 1.
```

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**